service/history/workflow/workflow_task_state_machine.go

```diff
@@ -568,10 +568,11 @@ func (m *workflowTaskStateMachine) processBuildIdRedirectInfo(
 	workflowTask *WorkflowTaskInfo,
 	redirectInfo *taskqueuespb.BuildIdRedirectInfo,
 ) (newWorkflowTask *WorkflowTaskInfo, converted bool, redirectCounter int64, err error) {
-	if !versioningStamp.GetUseVersioning() || versioningStamp.GetBuildId() == "" {
+	buildId := worker_versioning.BuildIdIfUsingVersioning(versioningStamp)
+	if buildId == "" && m.ms.IsStickyTaskQueueSet() {
```

Comment:
I'm thinking about the unversioned non-sticky case. Previously that would have hit this clause here and just returned no redirect.

Now... say buildId == "" and it's not a sticky queue. Then.. we don't do this, and we go on into validateBuildIdRedirectInfo. In there... we're expected to hit the `startedWorkerStamp.GetBuildId() == assignedBuildId` case (both empty string)?

I guess that works. Was that the intention?

Reply:
fixed.